File: ryan/week4/etc.py
def solution(land):
    dp = land
    for i in range(1, len(land)):
        for j in range(4):
            left = dp[i-1][:j]
            right = dp[i-1][j+1:]
            dp[i][j] += max(left + right)
    return max(dp[-1])

print(solution([[1,2,3,5],[5,6,7,8],[4,3,2,1]]))  # Output: 16

File: dongwoo/week4/etc.java
import java.util.*;
class Solution {
    int solution(int[][] land) {
        int answer = 0;
        for (int i = 1; i < land.length; i++) {
            for (int j = 0; j < 4; j++) {
                int max = 0;
                for (int x = 0; x < 4; x++) {
                    if (x != j) {
                        max = Math.max(max, land[i-1][x]);
                    }
                }
                land[i][j] += max;
            }
        }
        Arrays.sort(land[land.length-1]);
        return land[land.length-1][3];
    }
}
